### PR TITLE
Add new section to record Technical Decisions and added deployment pi…

### PR DESCRIPTION
**Pull request by mattempty**
…peline.

New section as discussed in Slack.

### CODE REVIEW

File: technical/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Form Builder - Technical Decisions
+
+THis is a central place to record any technical decisions that are not Archtiectual in nature.
```
Comment (tomas-stefano) [inline technical/README.md:3]: I think there is small fix here: THis

Comment (brenetic) [inline technical/README.md:3]: ~Archtiectual~ => architectural

File: technical/deployments-imagineering-session-2.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Deployments Imagineering session 2
+Created by Tomas Destefi 02 April 2020
+
+On the last two retros we talked about that our deployment needs a lot of improvements. So 02/04 Me, Brendan and Matt did the session 2 of the Deploy imagineering session.
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:4]: In the last two retros we talked about how our deployments need a lot of improvements
```suggestion
In the last two retros we talked about how our deployments need a lot of improvements. So 02/04 Me, Brendan and Matt did the session 2 of the Deploy imagineering session.
```

File: technical/deployments-imagineering-session-2.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+2. The runner app (application responsible to show the forms to the end user)
+3. The user datastore app (application responsible to handle the user data)
+4. The user filestore app (application responsible to handle user uploads)
+5. The pdf generator app (application responsible to generate pdfs for submitter)
+6. The service token cache app (application responsible to make all APIs to communicate with each other)
+7. The publisher app (app responsible of publishing the forms)
+
+## Benefits of this work
+When this work is finished we:
+1. We will have more confident on our deployments.
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:18]: ~confident on~ => confidence in

File: technical/deployments-imagineering-session-2.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+6. Faster Mean Time To Resolution.
+7. More test reliability.
+8. We can implement Continuous Delivery.
+
+## The first part of the work - Submitter
+We decided to start implementing the pipeline we envision on the submitter app.
+
+You can see step by step of the deployment pipeline (from left to right with some considerations).
+
+![alt text](TD01-deployment.png "Image of the deployment pipeline in context of theis document.")
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:32]: ~alt text~ => deployment pipeline ?

~theis~ => this

File: technical/deployments-imagineering-session-2.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+9. [https://trello.com/c/4hPTxwtd/410-add-healthcheck-step-to-submitter-production-environment](https://trello.com/c/4hPTxwtd/410-add-healthcheck-step-to-submitter-production-environment)
+10. [https://trello.com/c/qBo0i1wZ/411-add-smoke-tests-to-test-environment-after-the-deploy-production-is-complete](https://trello.com/c/qBo0i1wZ/411-add-smoke-tests-to-test-environment-after-the-deploy-production-is-complete)
+11. [https://trello.com/c/fwgQaChQ/412-alert-channel-about-submitter-pipeline-result](https://trello.com/c/fwgQaChQ/412-alert-channel-about-submitter-pipeline-result)
+12. [https://trello.com/c/nFlzcHv7/413-branches-can-deploy-to-test-on-submitter](https://trello.com/c/nFlzcHv7/413-branches-can-deploy-to-test-on-submitter)
+
+## After the work above is done
+After the work above is done we need to apply to other repositories like the Runner, the file store, datastore, and so on.
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:50]:
```suggestion
After the work above is done we need to apply the same changes to other repositories like the Runner, the file store, datastore, and so on.
```

File: technical/deployments-imagineering-session-2.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+## Considerations for the developers
+1. Don't overengineer but consider the same pipeline process to be applied to other repos like filestore, pdf generator so when doing for submitter the other could be easier to plug in.
+2. Branches can deploy to test (and stop there)
+3. Consider that we could leave CircleCI in the long term
+4. Branches can deploy to test (and stop there)
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:56]: 2 and 4 are the same?

File: technical/deployments-imagineering-session-2.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+5. In order to give us the confidence we need to test form via PDF/ CSVs / JSON endpoints (and submit each of those) (specific for submitter)
+6. After this work is started me (Tomas, Brendan and Matt) will review how is going what we agreed.
```
Comment (brenetic) [inline technical/deployments-imagineering-session-2.md:58]: ~me (Tomas, Brendan and Matt)~=> we  👑 😛

File: README.md [hunk 8]
```diff
@@ -6,10 +6,13 @@ It uses [JSON](documentation/glossary.md#json-and-json-schema) as its data inter
 
 - ## [Documentation](documentation/documentation.md)
 
-- ## [Architectual decision records](decisions/)
+- ## [Architectual decision records](decisions/README.md)
```
Comment (brenetic) [inline README.md:9]: ~Architectual~ => Architectural

File: technical/deployments-imagineering-session-2.md (hunk 2 above)
Comment (tomas-stefano) [reply technical/deployments-imagineering-session-2.md:4]: 🙈

**Pull request merged**